src/pool.jl

```diff
@@ -117,6 +175,20 @@ end
     end
 end
 
+@inline function Base.get!(pool::CategoricalPool{T}, level::CatValue) where T
+    pool === level.pool && return level.level
+    # Use invindex for O(1) lookup
+    # TODO: use a global table to cache this information for all pairs of pools
+    if level.pool.levels ⊈ keys(pool.invindex)
+        if isordered(pool)
+            throw(OrderedLevelsException(level, pool.levels))
+        end
+        newlevs, ordered = mergelevels(isordered(pool), pool.levels, level.pool.levels)
+        levels!(pool, newlevs)
```

Comment:
I understand that here we will get an error if `newlevs` element type is not consistent with `T`?

Reply:
Yes, we get a `MethodError` when conversion fails. See new test.